compiler/sempass2.nim

```diff
@@ -259,6 +276,29 @@ proc listGcUnsafety(s: PSym; onlyWarning: bool; conf: ConfigRef) =
   var cycleCheck = initIntSet()
   listGcUnsafety(s, onlyWarning, cycleCheck, conf)
 
+proc listSideEffects(result: var string; s: PSym; cycleCheck: var IntSet; conf: ConfigRef; indentLevel: int) =
+  template add(msg; level = indentLevel) =
+    result.add repeat("  ", level) & msg
```

Comment:
```nim
    result.add repeat(">", level) & " " & msg
```
characters are easier to count than spaces, and this would reuse the same rendering as used in `--processing` (refs https://github.com/nim-lang/Nim/pull/18372)

Reply:
What about using box symbols?
```
└─ lineinfo Error: reason
   ├─ lineinfo Hint: reason
   │  └─ lineinfo Hint: reason
   └─ lineinfo Hint: reason
```

Reply:
that'd be fine by me but seems harder to implement? it's ascii right?

Reply:
Not ascii. But could be:
```
`-- lineinfo Error: reason
    |-- lineinfo Hint: reason
    |   `-- lineinfo Hint: reason
    `-- lineinfo Hint: reason
```

I'll try to get to it in a later PR (maybe)

Reply:
ya let's defer to another PR, because my immediate reaction would be: "please generalize in an API that can be reused", even if in some private-for-now module

Reply:
Only whitespace before a diagnostic message is supported by the VSCode Nim plugin.

Relevant code:
https://github.com/pragmagic/vscode-nim/blob/0272a0544ddf147fde98f8a8f2b624b7dcb19eb3/src/nimBuild.ts#L86

I can submit a PR for it, but I'd first like to know for sure that leading '>' will be the standard for structured messages. This affects all tooling that parses the messages. I think I read somewhere that tooling support is a priority this year or something.

Reply:
i think we can move forward with this PR, and then in followup work either make vscode nim plugin be aware of `>`, or change back to ' ' here as last resort, but the rationale is that counting visibe chars (eg >) is easier.

There will likely be other static analysis messages done in the compiler that can be written this way so it's good to have 1 way to represent those.

I also wonder whether vscode plugin correctly handles things like the cyclic import error message, or other various multiline messages.

Reply:
I'm fine to move ahead.

The pragmagic/vscode-nim plugin doesn't seem to support much other than single line diagnostics. I'm not sure of the saem/vscode-nim plugin (just learned of it).

The mutation side effect message rendered by `varpartitions.$` also won't be parsed. That should maybe be combined with this side effect diagnostic message in  a later PR.